exotic/api/elca.py

```diff
@@ -247,140 +258,131 @@ def binner(arr, n, err=''):
         self.detrendederr = self.dataerr / self.airmass_model
         self.residuals = self.data - self.model
-        self.chi2 = np.sum(self.residuals**2/self.dataerr**2)
-        self.bic = len(self.bounds) * np.log(len(self.time)) - 2*np.log(self.chi2)
+        self.chi2 = np.sum(self.residuals ** 2 / self.dataerr ** 2)
+        self.bic = len(self.bounds) * np.log(len(self.time)) - 2 * np.log(self.chi2)
+
+        # compare fit chi2 to smoothed data chi2
+        dt = np.diff(np.sort(self.time)).mean()
+        si = np.argsort(self.time)
+        try:
+            self.sdata = savgol_filter(self.data[si], 1+2*int(0.5/24/dt), 2)
+        except:
+            self.sdata = np.ones(len(self.time))
+
+        schi2 = np.sum((self.data[si] - self.sdata)**2/self.dataerr[si]**2)
+        self.quality = schi2/self.chi2
+
+        # measured duration
+        tdur = (self.transit < 1).sum() * np.median(np.diff(np.sort(self.time)))
+
+        # test for partial transit
+        newtime = np.linspace(self.parameters['tmid']-0.2, self.parameters['tmid']+0.2, 10000)
+        newtran = transit(newtime, self.parameters)
+        masktran = newtran < 1
+        newdur = np.diff(newtime).mean()*masktran.sum()
+
+        self.duration_measured = tdur
+        self.duration_expected = newdur
 
     def fit_nested(self):
         freekeys = list(self.bounds.keys())
         boundarray = np.array([self.bounds[k] for k in freekeys])
-        bounddiff = np.diff(boundarray,1).reshape(-1)
+        bounddiff = np.diff(boundarray, 1).reshape(-1)
 
         def loglike(pars):
             # chi-squared
             for i in range(len(pars)):
                 self.prior[freekeys[i]] = pars[i]
             model = transit(self.time, self.prior)
-            model *= (self.prior['a1']*np.exp(self.prior['a2']*self.airmass))
-            return -0.5 * np.sum( ((self.data-model)/self.dataerr)**2 )
+            model *= np.exp(self.prior['a2'] * self.airmass)
+            detrend = self.data / model  # used to estimate a1
+            model *= np.median(detrend)
+            return -0.5 * np.sum(((self.data - model) / self.dataerr) ** 2)
 
+        @njit(fastmath=True)
         def prior_transform(upars):
             # transform unit cube to prior volume
-            return (boundarray[:,0] + bounddiff*upars)
+            return boundarray[:, 0] + bounddiff * upars
+
+        # TODO vectorize loglike
+        test = ReactiveNestedSampler(freekeys, loglike, prior_transform, log_dir=self.log_dir)
+        self.results = test.run(max_ncalls=int(self.max_ncalls))
 
-        dsampler = dynesty.DynamicNestedSampler(
-            loglike, prior_transform,
-            ndim=len(freekeys), bound='multi', sample='unif',
-            maxiter_init=5000, dlogz_init=1, dlogz=0.05,
-            maxiter_batch=100, maxbatch=10, nlive_batch=100
-        )
-        dsampler.run_nested(maxcall=1e6)
-        self.results = dsampler.results
+        # plots
+        test.plot()
```

Comment:
Is this going to cause any issues?

Reply:
test.plot() will give information on the run, including plots such as corner, trace and one for the run: 

[corner.pdf](https://github.com/rzellem/EXOTIC/files/6969615/corner.pdf)
[run.pdf](https://github.com/rzellem/EXOTIC/files/6969617/run.pdf)
[trace.pdf](https://github.com/rzellem/EXOTIC/files/6969618/trace.pdf)